lib/lightning/accounts.ex

```diff
@@ -70,17 +71,26 @@ defmodule Lightning.Accounts do
   def perform(%Oban.Job{args: %{"type" => "purge_deleted"}}) do
     users_with_activities =
       from(run in Lightning.Run,
-        join: user in Lightning.Accounts.User,
+        join: user in User,
         on: run.created_by_id == user.id,
         select: user.id,
         distinct: true
       )
 
+    users_with_files =
+      from(file in File,
+        join: user in User,
+        on: file.created_by_id == user.id,
+        select: user.id,
+        distinct: true
+      )
+
     users_to_delete =
       from(u in User,
         where:
           u.scheduled_deletion <= ago(0, "second") and
-            u.id not in subquery(users_with_activities)
+            u.id not in subquery(users_with_activities) and
+            u.id not in subquery(users_with_files)
```

Comment:
Given the recent performance improvement on deleting snapshots, can we just use `not exists` here as well? https://hexdocs.pm/ecto/Ecto.Query.API.html#exists/1

Reply:
well spot @midigofrank . I am getting a compile error with the parent_as to refer to the user. will get back to this during the weekend

Reply:
done

Reply:
done